File: datasets/IR-Plag/intellij-original-java-rest-kotlin/case-07/plagiarized/L2/04/multiarray.kt
package IR

import java.util.*

internal object multiarray {
    // 25779F8829AB7A7650E85A4CC871E6AC sangat ganteng
    @JvmStatic
    fun main(args: Array<String>) {
        val input = Scanner(System.`in`)
        print("Enter a 4 by 4 matrix row by row: ")
        val ma = Array(4) { DoubleArray(4) }
        //masukan array
        for (i in 0..3) for (j in 0..3) ma[i][j] = input.nextDouble()
        print("Sum of the elements in the major diagonal is " + sum(ma))
    }

    // menambahkan seluruh isi array
    fun sum(m: Array<DoubleArray>): Double {
        var sum = 0.0
        for (i in m.indices) sum += m[i][i]
        return sum
    }
}

File: datasets/IR-Plag/convertai-original-java-rest-python/case-07/plagiarized/L6/05/multiarray.py
import numpy as np

def sum(m):
    """
    Adds up all the elements on the main diagonal of the given 2D array.
    
    Args:
        m (numpy.ndarray): A 2D array.
        
    Returns:
        float: The sum of the elements on the main diagonal.
    """
    return np.sum(np.diag(m))

def main():
    print("Enter a 4 by 4 matrix row by row:")
    ma = np.zeros((4, 4))
    for i in range(4):
        for j in range(4):
            ma[3-i, 3-j] = float(input())
    
    sum_diag = sum(ma)
    print(f"Sum of the elements in the major diagonal is {sum_diag}")

if __name__ == "__main__":
    main()

